### PR TITLE
Fix openload "pigeons" problem

### PR DESCRIPTION
**Pull request by jose1711**
This change is [Reviewable](https://reviewable.io/reviews/kodi-czsk/script.module.stream.resolver/30)

### CODE REVIEW

File: lib/server/openload.py [hunk 1]
```diff
@@ -19,76 +19,254 @@
...
+        return re.escape(sValue)
+
+    def getNumberFromString(self, sValue):
+        sPattern = "\d+"
+        aMatches = re.findall(sPattern, sValue)
+        if (len(aMatches) > 0):
+            return aMatches[0]
+        return 0
+
+
```
Comment (lubo) [inline lib/server/openload.py:174]: These two classes looks like an unnecessary abstraction layer that we don't really need. And if we do so, you should rather implement it in `util` library. If I was you, I'd rewrite the resolver using existing libraries.

Comment (jose1711) [inline lib/server/openload.py:174]: The reason I am reluctant about a rewrite is that openload is a quickly moving target. Having the ability to keep pace with site updates by simply porting commits in LordVenom repo is a big plus for me.

File: lib/server/openload.py [hunk 2]
```diff
@@ -19,76 +19,254 @@
...
+    def request(self):
+        self.__sUrl = self.__sUrl.replace(' ', '+')
+        return self.__callRequest()
+
+    def getRequestUri(self):
+        return self.__sUrl + '?' + urllib.urlencode(self.__aParamaters)
+
+    def __setDefaultHeader(self):
+        self.addHeaderEntry('User-Agent', 'Mozilla/5.0 (Windows; U; Windows NT 5.1; de-DE; rv:1.9.0.3) Gecko/2008092417 Firefox/3.0.3')
```
Comment (lubo) [inline lib/server/openload.py]: This line is too long, trim it to maximum of 100 characters.

File: lib/server/openload.py [hunk 3]
```diff
@@ -19,76 +19,254 @@
...
+
+    def parseSingleResult(self, sHtmlContent, sPattern):
+        aMatches = re.compile(sPattern).findall(sHtmlContent)
+        if (len(aMatches) == 1):
+                aMatches[0] = self.__replaceSpecialCharacters(aMatches[0])
+                return True, aMatches[0]
+        return False, aMatches
+
+    def __replaceSpecialCharacters(self, sString):
+        return sString.replace('\\/','/').replace('&amp;','&').replace('\xc9','E').replace('&#8211;', '-').replace('&#038;', '&').replace('&rsquo;','\'').replace('\r','').replace('\n','').replace('\t','').replace('&#039;',"'")
```
Comment (lubo) [inline lib/server/openload.py]: This line is too long, trim it to maximum of 100 characters.

File: lib/server/openload.py (hunk 1 above)
Comment (lubo) [inline lib/server/openload.py:174]: All right then, that's a good argument.

**Comment by jose1711** [comment]
any update?

**Pull request merged**